### ISSUE TITLE
SANA-Sprint Student Training

### ISSUE THREAD

**Issue by nttung1110** [NVlabs/Sana#300]
Hi @lawrence-cj,

Thanks for your great work. I have tried the SANA-Sprint training code (with initialized student from SANA-Sprint Teacher 0.6B) using both implementation versions: the [original SANA repo](https://github.com/NVlabs/Sana/blob/main/train_scripts/train_scm_ladd.py), and [diffusers training version](https://github.com/huggingface/diffusers/blob/main/examples/research_projects/sana/train_sana_sprint_diffusers.py). Both experiments failed when the student prediction is quite noisy and does not produce high quality images. I didn't modify anything from the codebase except using a different training image-text pairs dataset with approximately 500k samples (with fixed resolution 1024x1024). The student model was trained in roughly 50 epochs with batch size = 8 and below is the output validation results:

<img width="720" height="288" alt="Image" src="https://github.com/user-attachments/assets/1613eb6d-1f70-42d4-a59d-e8373b3e3760" />

Questions: Do you have any tips or tricks during the training process? And do you think that the model still needs more training iterations to better converge? Appreciate your help!

**Comment by chenyiming-start** [comment]
Have you solved this problem? I ran into a similar problem when running diffusers training process. It seem that the gradient has converged, but the model checkpoints saved at all steps kept unchanged.

**Comment by chenyiming-start** [comment]
Hi, @lawrence-cj ,

I found two confusing parts in sana-sprint diffusers training demo, I hope this could be helpful to your work!

1. Attention processors in the [demo](https://github.com/huggingface/diffusers/blob/main/examples/research_projects/sana/train_sana_sprint_diffusers.py) are all turned into SanaVanillaAttnProcessor at line 1027, including the linear attention modules. Will this operation cause an error that the DiT in the training process differs from diffusers sanasprint pipline, and resulting in wrong training results?
2. The validation process in the [demo](https://github.com/huggingface/diffusers/blob/main/examples/research_projects/sana/train_sana_sprint_diffusers.py), line 1693,  specified the training transformer, which was defined as [SanaTrigFlow] at line 1067. However, the conversion from flow to trigflow has already been done in diffusers sanasprint pipline, and a repeated operation was performed in [SanaTrigFlow]. This might result in a mistake in validation results.
